[docs] update `handle` docs to clarify when it runs (#2208)
sveltejs/kit: documentation/docs/04-hooks.md

### handle

> This function runs on each endpoint request and page rendering, and determines the response. It receives the `request` object and a function called `resolve`, which invokes SvelteKit's router and generates a response accordingly. This allows you to modify response headers or bodies, or bypass SvelteKit entirely (for implementing endpoints programmatically, for example).

Reviewer: ```suggestion
This function runs every time SvelteKit responds to a request for a page or endpoint — whether that happens while the app is running, or during [prerendering](#ssr-and-javascript-prerender) — and determines the response. It receives the `request` object and a function called `resolve`, which invokes SvelteKit's router and generates a response accordingly. This allows you to modify response headers or bodies, or bypass SvelteKit entirely (for implementing endpoints programmatically, for example).

> Requests for static assets — which includes pages that were already prerendered — are _not_ handled by SvelteKit.
```

}
```

> > `handle` runs when a page is rendered. If a page is prerendered, `handle` will not run again when the user requests the page. If you would like to run some functionality on each request, see the documentation for your [adapter](#adapters).
> 

Reviewer: ```suggestion
```

Author: I think we still need the first sentence here. It's not at all obvious from the text above that `handle` doesn't run on prerendered pages. I'm fine with dropping the second sentence though

Author: Oh, nevermind. I missed that you added a callout above

}
```

> 

Author: ```suggestion
```

}
```

> > `handle` runs when a page is rendered. If a page is prerendered, `handle` will not run again when the user requests the page.

Author: ```suggestion
```

> > `handle` runs when a page is rendered. If a page is prerendered, `handle` will not run again when the user requests the page. If you would like to run some functionality on each request, see the documentation for your [adapter](#adapters).

Author: ```suggestion
> `handle` runs when a page is rendered. If a page is prerendered, `handle` will not run again when the user requests the page.
```
